<launch>

    <arg name="folder_path" doc="path into the folder with the images and the label file" />
    <arg name="label_file_name" doc="name of the yaml file containing the labels" default="labels.yaml" />
    <arg name="output_file_path" doc="path where the output file should be stored" default="$(arg folder_path)" />
    <arg name="loop_images" doc="looping the images to allow longer (but repetitive) tests" default="false" />
    <arg name="listen_balls" doc="listen for ball messages" default="true" />
    <arg name="listen_lines" doc="listen for line messages" default="true" />
    <arg name="listen_obstacles" doc="listen for obstacle messages" default="true" />
    <arg name="listen_goalposts" doc="listen for goalpost messages (not working)" default="true" />
    <arg name="listen_horizon" doc="listen for horizon messages" default="true" />

    <node pkg="bitbots_vision_tools" type="evaluator.py" name="bitbots_vision_evaluator" output="screen" >
        <rosparam command="load" file="$(find bitbots_vision_tools)/config/evaluator_params.yaml" />
        <param name="folder_path" value="$(arg folder_path)" />
        <param name="label_file_name" value="$(arg label_file_name)" />
        <param name="output_file_path" value="$(arg output_file_path)" />
        <param name="loop_images" value="$(arg loop_images)" />
        <param name="listen_balls" value="$(arg listen_balls)" />
        <param name="listen_lines" value="$(arg listen_lines)" />
        <param name="listen_obstacles" value="$(arg listen_obstacles)" />
        <param name="listen_goalposts" value="$(arg listen_goalposts)" />
        <param name="listen_horizon" value="$(arg listen_horizon)" />
    </node>
</launch>
